web/src/pages/api/post/search.ts

```diff
@@ -37,91 +39,72 @@ export default async function search(
-                      _count: "desc",
-                    },
-                  },
-                  { dateCreated: "desc" },
-                ],
-          select: {
-            id: true,
-            dateCreated: true,
-            prompt: true,
-            imageURL: true,
-            author: {
-              select: {
-                name: true,
-                image: true,
-                id: true,
+      // If a collectionId is passed, only return posts that are in that collection
+      collections: collectionId
+        ? {
+            some: {
+              id: collectionId,
+            },
+          }
+        : undefined,
+    },
+    take: limit ? parseInt(limit.toString()) : undefined,
+    skip: cursorId ? 1 : 0,
+    cursor: cursorId ? { id: cursorId.toString() } : undefined,
+    orderBy:
+      recentOnly === "true"
+        ? { dateCreated: "desc" }
+        : [
+            {
+              likes: {
+                _count: "desc",
               },
             },
-            likes: true,
-          },
+            { dateCreated: "desc" },
+          ],
+    select: {
+      id: true,
+      dateCreated: true,
+      prompt: true,
+      imageURL: true,
+      author: {
+        select: {
+          name: true,
+          image: true,
+          id: true,
         },
       },
-    });
-
-    if (searchedPosts?.posts) {
-      posts = searchedPosts.posts.filter((post: { prompt: string }) =>
-        post.prompt.toLowerCase().includes(search.toLowerCase())
-      );
-    }
-  } else {
-    posts = await prisma.post.findMany({
-      where: {
-        authorId: userId || undefined, // Leave undefined if no userId is passed
-        prompt: {
-          // Performs a case-insensitive search for prompts that contain the search string
-          contains: search,
-          mode: "insensitive",
+      _count: {
+        select: {
+          likes: true,
         },
       },
-      take: limit ? parseInt(limit.toString()) : undefined,
-      skip: cursorId ? 1 : 0,
-      cursor: cursorId ? { id: cursorId.toString() } : undefined,
-      orderBy:
-        recentOnly === "true"
-          ? { dateCreated: "desc" }
-          : [
-              {
-                likes: {
-                  _count: "desc",
-                },
-              },
-              { dateCreated: "desc" },
-            ],
-      select: {
-        id: true,
-        dateCreated: true,
-        prompt: true,
-        imageURL: true,
-        author: {
-          select: {
-            name: true,
-            image: true,
-            id: true,
-          },
+      // Get the current user's like status for each post
+      likes: {
+        where: {
+          userId: userId,
         },
-        likes: true,
       },
-    });
-  }
+    },
+  });
 
   res.json({
-    posts: posts ?? [],
+    posts:
+      posts.map((post) => ({
+        ...post,
+        _count: undefined, // Remove this field
+        likes: undefined, // Remove this field
```

Comment:
I prefer to express this idiom as
```ts
posts.map({ _count, likes, ...post }) => ({
  ...post,
```
because otherwise it create an enumerable property on the object

Reply:
ah yes, this is better for sure